torch/csrc/inductor/aoti_torch/shim_common.cpp

```diff
@@ -1291,3 +1295,120 @@ AOTITorchError aoti_torch_cpu__weight_int4pack_mm_cpu_tensor(
   });
 }
+
+// should TorchLibraryOpaque be a uniq_ptr?
+class TorchLibraryOpaque {
+  public:
+    TorchLibraryOpaque(torch::Library lib) : lib_(std::move(lib)) {}
+    
+    TorchLibraryOpaque(const TorchLibraryOpaque&) = delete;
+    TorchLibraryOpaque& operator=(const TorchLibraryOpaque&) = delete;
+    TorchLibraryOpaque(TorchLibraryOpaque&&) = default;
+    TorchLibraryOpaque& operator=(TorchLibraryOpaque&&) = default;
+    ~TorchLibraryOpaque() = default;
+  
+    torch::Library lib_;  // should this be private?
+};
+
+
+template <typename T>
+uint64_t from(T val) {
+  static_assert(sizeof(T) <= sizeof(uint64_t), "StableLibrary stack does not support parameter types larger than 64 bits.");
+  return *reinterpret_cast<uint64_t*>(&val);
+}
+
+template <typename T>
+T to(uint64_t val) {
+  return *reinterpret_cast<T*>(&val);
+}
+
+class StableIValueConverter: public c10::OperatorKernel {
+  public:
+    StableIValueConverter(void (*fn)(uint64_t*, int64_t, int64_t)) : fn_(fn) {}
+
+    void operator()(const c10::OperatorHandle& op, c10::DispatchKeySet keyset, torch::jit::Stack* stack) {
+      const auto& schema = op.schema();
+      const auto num_returns = schema.returns().size();
+      const auto num_arguments = schema.arguments().size();
+      // to make this faster, you can make this a C array on the stack --> though this may cause a stackoverflow
+      uint64_t *ministack = (uint64_t*)malloc((num_arguments + num_returns) * sizeof(uint64_t));
+      // std::unique_ptr<void *[]> ministack = std::make_unique<void*[]>(num_arguments + num_returns);
+
+      for (size_t idx = 0; idx < num_arguments; idx++) {  // rbarnes will prefer a c10::irange instead of this loop!
+        const c10::IValue& arg = torch::jit::peek(stack, idx, num_arguments);
+        if (arg.isInt()) {
+          TORCH_WARN("dealt with an Int");
+          ministack[idx] = from(arg.toInt());
+        } else if (arg.isDouble()) {
+          TORCH_WARN("dealt with a double");
+          ministack[idx] = from(arg.toDouble());
+        } else if (arg.isBool()) {
+          TORCH_WARN("dealt with a bool");
+          ministack[idx] = from(arg.toBool());
+        } else if (arg.isNone()) {
+          TORCH_WARN("dealt with a None");
+          ministack[idx] = from(nullptr);
+        } else if (arg.isTensor()) {
+          TORCH_WARN("dealt with a Tensor");
+          AtenTensorHandle ath = torch::aot_inductor::new_tensor_handle(std::move(const_cast<at::Tensor&>(arg.toTensor())));
+          ministack[idx] = from(ath);
+        } else {
+          TORCH_CHECK(false, "Other types of IValues not yet handled!");
+        }
+      }
+
+      // second function is going to take a stack of uint64_ts, cast them to our
+      // schema values, and run the function and modify the uint64_t stack
+      fn_(ministack, num_arguments, num_returns);
+
+      // now pop all inputs on stack. if we pop earlier, Tensors would go out of scope
+      // before calling the function
```

Comment:
I would not expect this to be true and that all the ATH going through the voidyvoid API are owning.